include/aikido/distance/JointAvoidanceConfigurationRanker.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef AIKIDO_DISTANCE_JOINTAVOIDANCECONFIGURATIONRANKER_HPP_
+#define AIKIDO_DISTANCE_JOINTAVOIDANCECONFIGURATIONRANKER_HPP_
+
+#include "aikido/distance/ConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+class JointAvoidanceConfigurationRanker : public ConfigurationRanker
+{
+public:
+  /// Constructor
+  ///
+  /// \param[in] metaSkeletonStateSpace Statespace of the skeleton.
+  /// \param[in] metaskeleton Metaskeleton of the robot.
+  JointAvoidanceConfigurationRanker(
+      statespace::dart::ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+      ::dart::dynamics::ConstMetaSkeletonPtr metaSkeleton);
+
+protected:
+  /// Returns cost as negative of distance from position limits.
+  double evaluateConfiguration(
+      const statespace::dart::MetaSkeletonStateSpace::State* solution)
+      const override;
+
+  /// Vector of indices corresponding to unbounded lower position limits.
+  std::vector<std::size_t> mUnboundedLowerLimitsIndices;
+
+  /// Vector of indices corresponding to unbounded upper position limits.
+  std::vector<std::size_t> mUnboundedUpperLimitsIndices;
+
+  /// (Modified) State corresponding to the lower position limits.
+  /// The positions at the indices in mUnboundedLowerLimitsIndices are modified
```

Comment:
I think we can change the descriptions for both lower/upper to clarify that the states are modified by `evaluateConfiguration` to match the input state before computing the distance from the lower limits.